python/dg_tools/leg_impedance_control/quad_leg_impedance_controller.py

```diff
@@ -165,36 +165,44 @@ def __init__(self, robot, ViconClientEntity, client_name = "vicon_client" , vico
         self.EntityName = EntityName
         self.client_name = client_name
         self.host_name_quadruped = vicon_ip
-        self.robot_vicon_name = "quadruped"
+        self.init_robot_properties()
 
-        self.vicon_client = ViconClientEntity(self.client_name)
+        self.vicon_client = ViconClientEntity('vicon_' + self.client_name)
         self.vicon_client.connect_to_vicon(self.host_name_quadruped)
         self.vicon_client.displaySignals()
 
         self.vicon_client.add_object_to_track("{}/{}".format(self.robot_vicon_name, self.robot_vicon_name))
 
         try:
             ## comment out if running on real robot
-            self.vicon_client.robot_wrapper(robot)
+            self.vicon_client.robot_wrapper(robot, self.robot_vicon_name)
         except:
             print("not in simulation")
 
 
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_position")
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_velocity_body")
-        self.robot.add_ros_and_trace(self.client_name, self.robot_vicon_name + "_velocity_world")
+        self.robot.add_ros_and_trace(self.vicon_client.name, self.robot_vicon_name + "_position")
```

Comment:
Be careful with this: the export to ROS is not really real-time safe. It is better t `add_to_trace` by default and let the user define the signal he wants to export to ros on the fly.